src/common/yamlwrapper.h

```diff
@@ -0,0 +1,79 @@
+/**
+*		This file is a part of rAthena++.
+*		Copyright(C) 2017 rAthena Development Team
+*		https://rathena.org - https://github.com/rathena
+*
+*	This program is free software: you can redistribute it and/or modify
+*	it under the terms of the GNU General Public License as published by
+*	the Free Software Foundation, either version 3 of the License, or
+*	(at your option) any later version.
+*
+*	This program is distributed in the hope that it will be useful,
+*	but WITHOUT ANY WARRANTY; without even the implied warranty of
+*	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*	GNU General Public License for more details.
+*
+*	You should have received a copy of the GNU General Public License
+*	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*
+*	Author(s)
+*		Jittapan Pluemsumran <secret@rathena.org>
+*/
+
+#pragma once
+#pragma warning(disable:4099) // Damn it, Microsoft! [Secret]
+
+#ifndef YAMLWRAPPER_H
+#define YAMLWRAPPER_H
+
+#ifdef __cplusplus
+#include "yaml-cpp/yaml.h"
+#include <string>
+
+extern "C" {
+#endif
+
+#include "cbasetypes.h"
+
+typedef struct yamlwrapper yamlwrapper;
+typedef struct yamliterator yamliterator;
+
+#ifdef __cplusplus
+class yamlwrapper {
```

Comment:
Does the typedef above not cause this to be translated to `class struct yamlwrapper`?
Also the typedefs should not be necessary if __cplusplus is defined or not?
This part is really confusing as hell x_x